operator/v1/types_insights.go

```diff
@@ -0,0 +1,124 @@
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// InsightsOperator holds cluster-wide information about the Insights Operator.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type InsightsOperator struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec is the specification of the desired behavior of the Insights.
+	// +kubebuilder:validation:Required
+	Spec InsightsOperatorSpec `json:"spec"`
+
+	// status is the most recently observed status of the Insights operator.
+	// +optional
+	Status InsightsOperatorStatus `json:"status"`
+}
+
+type InsightsOperatorSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+type InsightsOperatorStatus struct {
+	OperatorStatus `json:",inline"`
+	// gatherStatus provides basic information about the last Insights data gathering.
+	// When omitted, this means no data gathering has taken place yet.
+	// +optional
+	GatherStatus GatherStatus `json:"gatherStatus,omitempty"`
+	// reportStatus provides general Insights analysis results.
+	// +optional
+	ReportStatus ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatherStatus struct {
+	// lastGatherTime is the last time when Insights data gathering finished.
+	// +optional
+	LastGatherTime metav1.Time `json:"lastGatherTime,omitempty"`
+	// lastGatherDuration is the last data gathering duration.
+	// +optional
+	LastGatherDuration metav1.Duration `json:"lastGatherDuration,omitempty"`
+	// gatherers is a list of active gatherers (and their statuses) in the last gathering.
+	// +listType=atomic
+	// +optional
+	Gatherers []GathererStatus `json:"gatherers,omitempty"`
+}
+
+type ReportStatus struct {
+	// healthChecks provides basic information about active Insights health checks
+	// in a cluster.
+	// +listType=atomic
+	// +optional
+	HealthChecks []HealthCheck `json:"healthChecks,omitempty"`
+}
+
+type HealthCheck struct {
+	// description provides basic description of the healtcheck.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:MinLength=10
+	Description string `json:"description"`
+	// totalRisk of the healthcheck. Indicator of the total risk posed
+	// by the detected issue; combination of impact and likelihood. The values can be from 1 to 4,
+	// and the higher the number, the more important the issue.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=4
+	TotalRisk int32 `json:"totalRisk"`
+	// advisorURI provides the URL link to the Insights Advisor.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^https:\/\/\S+`
+	AdvisorURI string `json:"advisorURI"`
+	// state determines what the current state of the health check is.
```

Comment:
I think we should expand this comment to explain what the valid values are and that they can be changed by a user
```suggestion
	// state determines what the current state of the health check is.
	// Health checks can currently be either Enabled or Disabled.
	// By default, all health checks are ..., to ... a health check, do ...
```

Reply:
OK